cpk_lib_python_github/github_app_token_generator_package/github_app_token_generator/auth.py

```diff
@@ -0,0 +1,65 @@
+# -*- coding: utf-8 -*-
+"""GitHub App authentication utilities."""
+import logging
+import time
+
+import jwt
+
+logger = logging.getLogger(__name__)
+
+
+class GitHubAppAuth:
+    """Handle GitHub App authentication."""
+
+    def __init__(self, app_id: str, private_key: str):
+        self.app_id = app_id
+        self.private_key = private_key
+
+    def generate_jwt(self) -> str:
+        """Generate JWT token for GitHub App."""
+        try:
+            payload = {
+                "iat": int(time.time()),
+                "exp": int(time.time()) + (10 * 60),  # 10 minutes expiration
+                "iss": self.app_id,
+            }
+            token = jwt.encode(payload, self.private_key, algorithm="RS256")
+            logger.debug("Successfully generated JWT for app ID: %s", self.app_id)
+            return token
+        except Exception as error:
+            logger.error("Failed to generate JWT: %s", error)
+            raise ValueError(f"Failed to generate JWT: {error}") from error
+
+    @staticmethod
+    def read_private_key(private_key_path: str) -> str:
+        """Read private key from file."""
+        try:
+            with open(private_key_path, "r", encoding="utf-8") as key_file:
+                content = key_file.read()
+                logger.debug("Successfully read private key from: %s", private_key_path)
+                return content
+        except FileNotFoundError as error:
+            logger.error("Private key file not found: %s", private_key_path)
+            raise error
+        except IOError as error:
+            logger.error("Error reading private key file: %s", error)
+            raise error
+
+    @staticmethod
+    def get_private_key_content(private_key_path=None, private_key_content=None) -> str:
```

Comment:
```suggestion
    def get_private_key_content(private_key_path: Optional[str] = None,
                            private_key_content: Optional[str] = None) -> str:
```

since both private_key_path and private_key_content are optional you can do this 

if you accepting this, you need 
`from typing import Optional`

Reply:
Addressed